src/rust/src/merkle_frontier.rs

```diff
@@ -0,0 +1,161 @@
+use core::mem::size_of_val;
+use core::pin::Pin;
+
+use incrementalmerkletree::{bridgetree, Altitude, Frontier, Hashable};
+use orchard::{bundle::Authorized, tree::MerkleHashOrchard};
+use tracing::error;
+use zcash_primitives::{
+    merkle_tree::{
+        incremental::{read_frontier_v1, write_frontier_v1},
+        CommitmentTree, HashSer,
+    },
+    transaction::components::Amount,
+};
+
+use crate::{streams::CppStream, wallet::Wallet};
+
+pub const MERKLE_DEPTH: u8 = 32;
+
+#[cxx::bridge]
+mod ffi {
+    extern "C++" {
+        include!("streams.h");
+
+        #[cxx_name = "RustDataStream"]
+        type RustStream = crate::streams::ffi::RustStream;
+    }
+
+    #[namespace = "merkle_frontier"]
+    extern "Rust" {
+        type Orchard;
```

Comment:
I'm a bit skeptical of having this just be called `Orchard` and relying on the namespace to provide more clarity for its meaning, given that the namespace also contains `OrchardBundle` and `OrchardWallet`

Reply:
`OrchardBundle` and `OrchardWallet` are temporary helpers, only present while our FFI is partially converted to `cxx`. They will be removed eventually.